### ISSUE TITLE
Cron tailgate-like workflow triggering (start a workflow in some time after it completes and not according to cron schedule) [Java]

### ISSUE THREAD

**Issue by pvolchak** [temporalio/temporal#1994]
**Is your feature request related to a problem? Please describe.**
Currently there is no way to trigger a workflow in some time. There is support for cron workflows but if the workflow is taking up the whole time bracket then the workflow runs might be immediate.

**Describe the solution you'd like**
Let's consider a scenario of polling a lot of data from multiple data sources. I start a cron workflow that is supposed to run with a 10 minute interval. Let's say the workflow started it's execution. It completes pulling all the data in 9 minutes. Then cron triggers it again in 1 minute instead of in 10 minutes, which is correct, but not a desired behavior.

**Describe alternatives you've considered**
Using some other cron or job mechanisms to identify when to start a non cron workflow

**Comment by dnr** [comment]
You're right that the behavior you want isn't possible with the current cron feature. We're working on redesigning that feature and will definitely take this use case into account.

In the meantime, there are a few ways to achieve that functionality with the current features:

You can use a simple loop: wrap the repeated logic followed by a sleep in a for loop that runs 100 times. After 100 or 1000 times, do continue-as-new to restart the workflow. The repeated logic could be included in that workflow itself, or as a child workflow that the parent starts and waits on.

Alternatively, put a sleep at the beginning of the workflow and use continue-as-new each time.

Make sure the workflow has an infinite execution and run timeout (the default). You could also add a workflow retry policy to ensure that it keeps running if the workflow somehow fails.

**Comment by pvolchak** [comment]
the problem that i am facing with continue-as-new is its status. maybe its already in the working, but any time i see the status continue-as-new i have no idea if it gets to completed/error state, at least in the cron flow of things. Do you know of a way to fix/correct/modify that?

**Comment by dnr** [comment]
If you define an outer workflow that just does a loop around a child workflow plus sleep, and put your logic in the child workflow, you'll be able to see the status of each run. (The outer workflow will always be "running" or "continued as new".) You could even adjust the sleep time based on the result of the child if you want.

**Comment by pvolchak** [comment]
that is fair. i will play with that. thanks a lot.

**Comment by yiminc** [comment]
Cron has been changed since 1.14 and the workflow is now complete with it actual state and not as continue-as-new.

**Comment by dnr** [comment]
The original request is a valid use case that isn't addressed by the current cron feature: run a workflow repeatedly, sleeping for a fixed period in between runs.

I think we can address this in the new experience, so I'll keep it open until then (or we decide that a loop + child workflow is the best solution).
